.scripts/symlink_ds.sh

```diff
@@ -5,25 +5,26 @@ IFS=$'\n\t'
 symlink_ds() {
     run_script 'set_permissions' "${SCRIPTNAME}"
 
-    # /usr/bin/ds
-    if [[ -L "/usr/bin/ds" ]] && [[ ${SCRIPTNAME} != "$(readlink -f /usr/bin/ds)" ]]; then
-        info "Attempting to remove /usr/bin/ds symlink."
-        sudo rm -f "/usr/bin/ds" || fatal "Failed to remove file.\nFailing command: ${F[C]}sudo rm -f \"/usr/bin/ds\""
-    fi
-    if [[ ! -L "/usr/bin/ds" ]]; then
-        info "Creating /usr/bin/ds symbolic link for DockSTARTer."
-        sudo ln -s -T "${SCRIPTNAME}" /usr/bin/ds || fatal "Failed to create symlink.\nFailing command: ${F[C]}sudo ln -s -T \"${SCRIPTNAME}\" /usr/bin/ds"
+    if findmnt -n /usr | grep "ro" > /dev/null; then
+        if [[ $PATH != *"$HOME/bin"* ]]; then
+            warn "Read only /usr filesystem detected. Symlinks will be created in $HOME/bin. You will need to add this to your path."
+        fi
+        mkdir -p "$HOME/bin" # Make sure the path exists.
+        ds_symlink_targets=("HOME/bin")
+    else
+        ds_symlink_targets=("/usr/bin/ds" "/usr/local/bin/ds")
     fi
 
-    # /usr/local/bin/ds
-    if [[ -L "/usr/local/bin/ds" ]] && [[ ${SCRIPTNAME} != "$(readlink -f /usr/local/bin/ds)" ]]; then
-        info "Attempting to remove /usr/local/bin/ds symlink."
-        sudo rm -f "/usr/local/bin/ds" || fatal "Failed to remove file.\nFailing command: ${F[C]}sudo rm -f \"/usr/local/bin/ds\""
-    fi
-    if [[ ! -L "/usr/local/bin/ds" ]]; then
-        info "Creating /usr/local/bin/ds symbolic link for DockSTARTer."
-        sudo ln -s -T "${SCRIPTNAME}" /usr/local/bin/ds || fatal "Failed to create symlink.\nFailing command: ${F[C]}sudo ln -s -T \"${SCRIPTNAME}\" /usr/local/bin/ds"
-    fi
+    for target in "${ds_symlink_targets[@]}"; do
+        if [[ -L ${target} ]] && [[ ${SCRIPTNAME} != "$(readlink -f "${target}")" ]]; then
+            info "Attempting to remove ${target} symlink."
+            sudo rm -f "${target}" || fatal "Failed to remove file. Failing command: sudo rm -f \"${target}\""
```

Comment:
```suggestion
            sudo rm -f "${target}" || fatal "Failed to remove file.\nFailing command: ${F[C]}sudo rm -f \"${target}\""
```
Keep the newline and color variable